crates/router/src/connector/braintree/braintree_graphql_transformers.rs

```diff
@@ -0,0 +1,988 @@
+            let edge_data = item
+                .response
+                .data
+                .as_ref()
+                .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?
+                .search
+                .as_ref()
+                .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?
+                .refunds
+                .as_ref()
+                .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?
+                .edges
+                .first()
+                .ok_or(errors::ConnectorError::MissingConnectorRefundID)?;
+            let connector_refund_id = &edge_data.node.id;
+            let response = Ok(types::RefundsResponseData {
+                connector_refund_id: connector_refund_id.to_string(),
+                refund_status: enums::RefundStatus::from(edge_data.node.status.clone()),
+            });
+            Ok(Self {
+                response,
+                ..item.data
+            })
+        }
+    }
+}
+
+#[derive(Default, Debug, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreditCardData {
+    number: cards::CardNumber,
+    expiration_year: Secret<String>,
+    expiration_month: Secret<String>,
+    cvv: Secret<String>,
+    cardholder_name: Secret<String>,
+}
+
+#[derive(Default, Debug, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct InputData {
+    client_mutation_id: String,
+    credit_card: CreditCardData,
+}
+
+#[derive(Default, Debug, Clone, Serialize)]
+pub struct VariableInput {
+    input: InputData,
+}
+
+#[derive(Default, Debug, Clone, Serialize)]
+pub struct BraintreeTokenRequest {
+    query: String,
+    variables: VariableInput,
+}
+
+impl TryFrom<&types::TokenizationRouterData> for BraintreeTokenRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::TokenizationRouterData) -> Result<Self, Self::Error> {
+        match item.request.payment_method_data.clone() {
+            api::PaymentMethodData::Card(card_data) => {
+                let query = "mutation  tokenizeCreditCard($input: TokenizeCreditCardInput!) { tokenizeCreditCard(input: $input) { clientMutationId paymentMethod { id } } }".to_string();
+                let input = InputData {
+                    client_mutation_id: "12345667890".to_string(),
+                    credit_card: CreditCardData {
+                        number: card_data.card_number,
+                        expiration_year: card_data.card_exp_year,
+                        expiration_month: card_data.card_exp_month,
+                        cvv: card_data.card_cvc,
+                        cardholder_name: card_data.card_holder_name,
+                    },
+                };
+                Ok(Self {
+                    query,
+                    variables: VariableInput { input },
+                })
+            }
+            _ => Err(errors::ConnectorError::NotImplemented("Payment Method".to_string()).into()),
```

Comment:
Don't use default throw NotImplemented for the payment methods which are supported by connector and not implemented in our system. throw NotSupported for payment method which are not supported by the connector